bats/tests/containers/published-udp-ports.bats

```diff
@@ -0,0 +1,42 @@
+load '../helpers/load'
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+run_container_with_published_udp_port_and_connect() {
+    local ip=$1
+    local port=$2
+    local netcat_connect_addr=$3
+    ctrctl run -d --name socat-udp-"$port" -p "$ip":"$port":"$port"/udp alpine sh -c \
+        "apk add --no-cache socat && socat -v -T1 UDP-RECVFROM:$port,fork STDOUT"
+    sleep 5
```

Comment:
Is the sleep for `apk add`? Might be better to build a temporary image instead (so that crappy internet doesn't cause the test to fail).

Reply:
Yeah, we should not use `sleep` in tests without having a check at then end to verify that whatever we were waiting for actually happened. Otherwise we'll spend a lot of time debugging race conditions.

Reply:
The original `sleep` was intended to give the container time to start and allow netcat to connect. While we have sleep in other BATS tests, I agree that relying on arbitrary delays could be problematic.

I've changed the test by building a custom image based on Alpine with socat preinstalled, which avoids the need for installation at runtime. Additionally, I've replaced the fixed sleep with a retry around the `nc` command.